subs/pantry/src/Pantry/SQLite.hs

```diff
@@ -69,33 +71,84 @@ initStorage description migration fp inner = do
 -- above.
 withWriteLock
   :: HasLogFunc env
-  => Path Abs File -- ^ SQLite database file
+  => Utf8Builder -- ^ database description, for lock messages
+  -> Path Abs File -- ^ SQLite database file
   -> RIO env a
   -> RIO env a
-withWriteLock dbFile inner = do
+withWriteLock desc dbFile inner = do
   let lockFile = toFilePath dbFile ++ ".pantry-write-lock"
   withRunInIO $ \run -> do
     mres <- withTryFileLock lockFile Exclusive $ const $ run inner
     case mres of
       Just res -> pure res
       Nothing -> do
-        run $ logInfo "Unable to get a write lock on the Pantry database, waiting..."
-        shouldStopComplainingVar <- newTVarIO False
-        let complainer = fix $ \loop -> do
-              delay <- registerDelay $ 60 * 1000 * 1000 -- 1 minute
-              shouldComplain <-
-                atomically $
-                  -- Delay has triggered, time to complain again
-                  (readTVar delay >>= checkSTM >> pure True) <|>
-                  -- Time to stop complaining, ignore that delay immediately
-                  (readTVar shouldStopComplainingVar >>= checkSTM >> pure False)
-              when shouldComplain $ do
-                run $ logWarn "Still waiting on the Pantry database write lock..."
-                loop
-            stopComplaining = atomically $ writeTVar shouldStopComplainingVar True
-            worker = withFileLock lockFile Exclusive $ const $ do
-              run $ logInfo "Acquired the Pantry database write lock"
-              stopComplaining
-              run inner
-        runConcurrently $ Concurrently complainer
-                       *> Concurrently (worker `finally` stopComplaining)
+        let complainer :: Talker IO
+            complainer delay = run $ do
+              -- Wait five seconds before giving the first message to
+              -- avoid spamming the user for uninteresting file locks
+              delay $ 5 * 1000 * 1000 -- 5 seconds
+              logInfo $ "Unable to get a write lock on the " <> desc <> " database, waiting..."
+
+              -- Now loop printing a message every 1 minute
+              forever $ do
+                delay (60 * 1000 * 1000) -- 1 minute
+                  `onException` logInfo ("Acquired the " <> desc <> " database write lock")
```

Comment:
This is minor, and not sure if it's worth the trouble, but it might be a good idea to do `withException` and only print the message about acquired lock whenever it is a `DoneTalking` exception. Otherwise upon async exception it will also print that it has acquired the database lock, which can be misleading.